cromshell

```diff
@@ -0,0 +1,311 @@
+SCRIPTDIR="$( cd "$( dirname "${UNALIASED_SCRIPT_NAME}" )" && pwd )"
+SCRIPTNAME=$( echo $0 | sed 's#.*/##g' )
+MINARGS=1
+MAXARGS=99
+
+################################################################################
+
+# read env var, or use default server URL
+export CROMWELL_URL=${CROMWELL_URL:-"https://cromwell-v30.dsde-methods.broadinstitute.org"}
+
+CROMSHELL_CONFIG=${HOME}/.cromshell
+mkdir -p ${CROMSHELL_CONFIG}
+
+CROMWELL_METADATA_PARAMETERS="excludeKey=submittedFiles"
+CROMWELL_LAST_WORKFLOW_FILE="${CROMSHELL_CONFIG}/last.workfow.id"
+CROMWELL_DIARY_WORKFLOW_FILE="${CROMSHELL_CONFIG}/all.workfow.ids"
+
+################################################################################
+
+SUB_COMMAND=''
+WORKFLOW_ID=''
+OPTIONS=''
+
+################################################################################
+
+function simpleUsage()
+{
+  echo -e "Usage:    ${SCRIPTNAME} <subcommand> [options]"
+  echo -e "Pokes Cromwell REST endpoints at a server specified by the "
+  echo -e "environment variable \$CROMWELL_URL" 
+}
+
+#Define a usage function:
+function usage() 
+{
+  simpleUsage
+  echo
+  echo -e "When submitting a workflow, the returned worfklow-id will become the" 
+  echo -e "last used workflow id. Also, you can use -n instead of a workflow-id to "
+  echo -e "use a previous run (status -1 would be the status of the last run," 
+  echo -e "-2 would be the one before that).  However, for commands that "
+  echo -e "accepts multiple workflow-id's (status and abort), this does not apply, "
+  echo -e "the explicit workflow-id's are expected if you provide multiple ones."
+  echo
+	echo -e "NOTE: As a convenience, if you omit a workflow-id from a command,"
+ 	echo -e "the last used workflow-id will be used as a default.  "
+  echo
+  echo -e "Requires the following programs to be installed:"
+  echo -e "  curl"
+  echo -e "  mail"
+  echo -e "  jq"
+  echo
+  echo -e "Supported Subcommands:"
+  echo -e "   submit [wdl-file] [inputs] [options] [dependencies]"
+  echo -e "   status [worfklow-id] [[worfklow-id]...]"
+  echo -e "   logs [worfklow-id] [[worfklow-id]...]"
+  echo -e "   metadata [worfklow-id] [[worfklow-id]...]"
+  echo -e "   slim-metadata [worfklow-id] [[worfklow-id]...]"
+  echo -e "   execution-status-count [worfklow-id] [[worfklow-id]...]"
+  echo -e "   timing [worfklow-id] [[worfklow-id]...]"
+  echo -e "   abort [worfklow-id] [[worfklow-id]...]"
+	echo -e "   notify [workflow-id] email_address"
+  echo 
+  echo -e "Return values:"
+  echo -e "  0                  SUCCESS"
+	echo -e "  ANYTHING_BUT_ZERO  FAILURE/ERROR"
+  echo
+}
+
+#Display a message to std error:
+function error() 
+{
+  echo "$1" 1>&2 
+}
+
+TMPFILELIST=''
+function makeTemp()
+{
+  local f
+  f=$( mktemp )
+  TMPFILELIST="${TMPFILELIST} $f"
+  echo $f
+}
+
+function cleanTempVars()
+{
+  rm -f ${TMPFILELIST}
+}
+
+function at_exit()
+{
+  cleanTempVars
+}
+
+function checkPipeStatus() {
```

Comment:
oh, neat, I didn't know about that.

Reply:
Yeah - it's VERY helpful when checking whether a chain of commands has passed (and really the only way).